File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/brev/HentMuligeBrevmottakereRequest.java
package no.nav.melosys.tjenester.gui.dto.brev;

import no.nav.melosys.domain.kodeverk.brev.Produserbaredokumenter;
import no.nav.melosys.service.brev.bestilling.HentMuligeBrevmottakereService;

public record HentMuligeBrevmottakereRequest(Produserbaredokumenter produserbartdokument, String orgnr) {
    public HentMuligeBrevmottakereService.RequestDto tilHentMuligeBrevmottakereRequestDto(Long behandlingID) {
        return new HentMuligeBrevmottakereService.RequestDto(this.produserbartdokument(), behandlingID, this.orgnr());
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/brev/HentMuligeBrevmottakereRequest.kt
package no.nav.melosys.tjenester.gui.dto.brev

import no.nav.melosys.domain.kodeverk.brev.Produserbaredokumenter
import no.nav.melosys.service.brev.bestilling.HentMuligeBrevmottakereService

@JvmRecord
data class HentMuligeBrevmottakereRequest(
    @JvmField val produserbartdokument: Produserbaredokumenter,
    @JvmField val orgnr: String,
    @JvmField val institusjonID: String
) {
    fun tilHentMuligeBrevmottakereRequestDto(behandlingID: Long?): HentMuligeBrevmottakereService.RequestDto {
        return HentMuligeBrevmottakereService.RequestDto(
            this.produserbartdokument,
            behandlingID!!,
            this.orgnr,
            this.institusjonID
        )
    }
}
